### ISSUE TITLE
reenable and debug test flake in test/integration/controller_instance_test.go:TestCreateServiceInstanceWithProvisionFailure

### ISSUE THREAD

**Issue by carolynvs** [kubernetes-incubator/service-catalog#2036]
I've seen our builds randomly fail on this test a few times for no discernable reason. If anyone has time to look into it, that would be incredibly helpful!

```
--- FAIL: TestCreateServiceInstanceWithProvisionFailure/network_timeout (6.36s)
    	framework.go:111: Server started on port 36468
    	framework.go:142: Test client will use API Server URL of https://localhost:36468
    	controller_test.go:787: controller start
    	controller_test.go:1420: unexpected action type: expected ProvisionInstance, got DeprovisionInstance
    	framework.go:116: Shutting down server on port: 36468
```

**Comment by teaguecole** [comment]
For PR #2055, After pushing commit 447debe, I ran in to this issue.

**Comment by MHBauer** [comment]
Slightly different test failure on #2087
```
[Build & Unit/Integration Tests]     --- FAIL: TestCreateServiceInstanceWithProvisionFailure/Status_Created (7.92s)
[Build & Unit/Integration Tests]     	framework.go:111: Server started on port 39737
[Build & Unit/Integration Tests]     	framework.go:142: Test client will use API Server URL of https://localhost:39737
[Build & Unit/Integration Tests]     	controller_test.go:790: controller start
[Build & Unit/Integration Tests]     	controller_test.go:1426: unexpected action type: expected ProvisionInstance, got DeprovisionInstance
[Build & Unit/Integration Tests]     	framework.go:116: Shutting down server on port: 39737
```

**Comment by MHBauer** [comment]
from #2099 
```
[Build & Unit/Integration Tests] === RUN   TestCreateServiceInstanceWithProvisionFailure/other_2xx
[Build & Unit/Integration Tests] []fake.Action{fake.Action{Type:"GetCatalog", Request:interface {}(nil)}, fake.Action{Type:"ProvisionInstance", Request:(*v2.ProvisionRequest)(0xc42012dce0)}, fake.Action{Type:"DeprovisionInstance", Request:(*v2.DeprovisionRequest)(0xc421aac6c0)}, fake.Action{Type:"DeprovisionInstance", Request:(*v2.DeprovisionRequest)(0xc421cb5c80)}, fake.Action{Type:"DeprovisionInstance", Request:(*v2.DeprovisionRequest)(0xc421aad140)}, fake.Action{Type:"ProvisionInstance", Request:(*v2.ProvisionRequest)(0xc422f89d50)}, fake.Action{Type:"ProvisionInstance", Request:(*v2.ProvisionRequest)(0xc4202e3730)}, fake.Action{Type:"DeprovisionInstance", Request:(*v2.DeprovisionRequest)(0xc4212eb900)}}E0606 20:33:16.120293   11187 watcher.go:208] watch chan error: rpc error: code = Canceled desc = context canceled
[Build & Unit/Integration Tests] E0606 20:33:16.122686   11187 status.go:64] apiserver received an error that is not an metav1.Status: rpc error: code = Unavailable desc = transport is closing
[Build & Unit/Integration Tests] E0606 20:33:16.126487   11187 controller_instance.go:1243] ServiceInstance "test-namespace/test-instance" v1458: Failed to update status: rpc error: code = Unavailable desc = transport is closing
[Build & Unit/Integration Tests] === RUN   TestCreateServiceInstanceWithProvisionFailure/3XX
...
[Build & Unit/Integration Tests]     --- FAIL: TestCreateServiceInstanceWithProvisionFailure/other_2xx (7.12s)
[Build & Unit/Integration Tests]     	framework.go:111: Server started on port 38716
[Build & Unit/Integration Tests]     	framework.go:142: Test client will use API Server URL of https://localhost:38716
[Build & Unit/Integration Tests]     	controller_test.go:790: controller start
[Build & Unit/Integration Tests]     	controller_test.go:1426: unexpected action type: expected ProvisionInstance, got DeprovisionInstance
[Build & Unit/Integration Tests]     	framework.go:116: Shutting down server on port: 38716
```
```

**Comment by MHBauer** [comment]
from #2115 
```
[Build & Unit/Integration Tests] === RUN   TestCreateServiceInstanceWithProvisionFailure/network_timeout
[Build & Unit/Integration Tests] []fake.Action{fake.Action{Type:"GetCatalog", Request:interface {}(nil)}, fake.Action{Type:"ProvisionInstance", Request:(*v2.ProvisionRequest)(0xc42372db90)}, fake.Action{Type:"DeprovisionInstance", Request:(*v2.DeprovisionRequest)(0xc422322fc0)}, fake.Action{Type:"DeprovisionInstance", Request:(*v2.DeprovisionRequest)(0xc42005bb00)}, fake.Action{Type:"DeprovisionInstance", Request:(*v2.DeprovisionRequest)(0xc422323b00)}, fake.Action{Type:"ProvisionInstance", Request:(*v2.ProvisionRequest)(0xc420ec57a0)}, fake.Action{Type:"ProvisionInstance", Request:(*v2.ProvisionRequest)(0xc42102be30)}, fake.Action{Type:"DeprovisionInstance", Request:(*v2.DeprovisionRequest)(0xc420aa5740)}}E0615 14:40:25.605321   11200 watcher.go:208] watch chan error: rpc error: code = Canceled desc = context canceled
[Build & Unit/Integration Tests] E0615 14:40:25.610781   11200 controller_instance.go:1262] ServiceInstance "test-namespace/test-instance" v1622: Failed to update status: Put https://localhost:54419/apis/servicecatalog.k8s.io/v1beta1/namespaces/test-namespace/serviceinstances/test-instance/status: dial tcp 127.0.0.1:54419: getsockopt: connection refused

....

[Build & Unit/Integration Tests]     --- FAIL: TestCreateServiceInstanceWithProvisionFailure/network_timeout (7.25s)
[Build & Unit/Integration Tests]     	framework.go:111: Server started on port 54419
[Build & Unit/Integration Tests]     	framework.go:142: Test client will use API Server URL of https://localhost:54419
[Build & Unit/Integration Tests]     	controller_test.go:790: controller start
[Build & Unit/Integration Tests]     	controller_test.go:1426: unexpected action type: expected ProvisionInstance, got DeprovisionInstance
[Build & Unit/Integration Tests]     	framework.go:116: Shutting down server on port: 54419
```

**Comment by MHBauer** [comment]
Entire test disabled by #2143, need to review the necessity of the test.

**Comment by MHBauer** [comment]
closed by #2153
